### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by EidHachem**
### In this PR we did the following:

- Added unit tests to these classes:
   - Book
   - Person
   - Student
   - Teacher
   - Classroom
   - Rental
   - the Decorator design pattern

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    it 'should return true if he can use service' do
+      expect(patrick.can_use_services?).to eq true
+    end
+
+    it 'should add into rentals' do
+      eid.add_rental(eid)
+      expect(eid.rentals.length).to eq 1
+    end
+  end
```
Comment (TedLivist) [inline spec/person_spec.rb]: - [x] Kindly include test(s) that evaluate the constructor method for creating a person.

  For example: Does it correctly return the person's age, name? Does it initialize an empty `rentals` array? In a scenario where no name attribute was passed, does it return `"Unknown"`? And so on.

File: spec/student_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  context 'When adding a new student' do
+    student = Student.new(25, 'Patrick', parent_permission: true)
+
+    it 'creates a new student' do
+      expect(student).to be_an_instance_of Student
+    end
+
+    it 'Student can use services' do
+      expect(student.can_use_services?).to eq true
+    end
```
Comment (TedLivist) [inline spec/student_spec.rb:13]: - [x] We need to test all public methods. Kindly include test for the `play_hooky` method contained in the Student class.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @EidHachem and @Pazzo97,

Good job so far!
✔️ Good documentation
✔️ Available well-written tests are all passing
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
Hi @EidHachem 🙋‍♂️, 

Thanks for making the requested change 👍

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**